src/core/a-mixin.js

```diff
@@ -9,106 +9,97 @@ var MULTIPLE_COMPONENT_DELIMITER = '__';
  * @member {object} componentCache - Cache of pre-parsed values. An object where the keys
  *         are component names and the values are already parsed by the component.
  */
-module.exports = registerElement('a-mixin', {
-  prototype: Object.create(ANode.prototype, {
-    createdCallback: {
-      value: function () {
-        this.componentCache = {};
-        this.id = this.getAttribute('id');
-        this.isMixin = true;
-      }
-    },
+class AMixin extends ANode {
+  static get observedAttributes () {
+    return Object.keys(components);
+  }
 
-    attributeChangedCallback: {
-      value: function (attr, oldVal, newVal) {
-        this.cacheAttribute(attr, newVal);
-        this.updateEntities();
-      }
-    },
+  constructor (self) {
+    super(self);
+    self = self || this;
+    self.componentCache = {};
+    self.isMixin = true;
+  }
 
-    attachedCallback: {
-      value: function () {
-        this.sceneEl = this.closestScene();
-        this.cacheAttributes();
-        this.updateEntities();
-        this.load();
-      }
-    },
+  attributeChangedCallback (attr, oldVal, newVal) {
+    super.attributeChangedCallback(attr, oldVal, newVal);
+    this.cacheAttribute(attr, newVal);
+    this.updateEntities();
+  }
 
-    /**
-     * setAttribute that parses and caches component values.
-     */
-    setAttribute: {
-      value: function (attr, value) {
-        window.HTMLElement.prototype.setAttribute.call(this, attr, value);
-        this.cacheAttribute(attr, value);
-      }
-    },
+  connectedCallback () {
+    this.id = this.getAttribute('id');
+    this.sceneEl = this.closestScene();
+    this.cacheAttributes();
+    this.updateEntities();
+    this.load();
+  }
 
-    /**
-     * If `attr` is a component, then parse the value using the schema and store it.
-     */
-    cacheAttribute: {
-      value: function (attr, value) {
-        var component;
-        var componentName;
+  /**
+   * setAttribute that parses and caches component values.
+   */
+  setAttribute (attr, value) {
+    window.HTMLElement.prototype.setAttribute.call(this, attr, value);
```

Comment:
You can call `super.setAttribute(attr, value)` without reaching to the prototype directly, but you might also consider doing this in `attributeChangedCallback` if you can.

Reply:
We try to avoid `attributeChangedCallback` as much as as possible to reduce asynchrony.

Reply:
In this case we don't want the `a-node` setAttribute logic to run but we still want the new values to be reflected in the DOM